Formula/spack.rb

```diff
@@ -0,0 +1,45 @@
+class Spack < Formula
+  desc "Package manager that builds multiple versions and configurations of software"
+  homepage "https://spack.io"
+  url "https://github.com/spack/spack/archive/v0.16.0.tar.gz"
+  sha256 "064b2532c70916c7684d4c7c973416ac32dd2ea15f5c392654c75258bfc8c6c2"
+  license any_of: ["Apache-2.0", "MIT"]
+  head "https://github.com/spack/spack.git", branch: "develop"
+
+  bottle :unneeded
+
+  depends_on "python@3.9"
+
+  def install
+    cp_r Dir["bin", "etc", "lib", "share", "var"], prefix.to_s
+  end
+
+  test do
+    system "#{bin}/spack", "--version"
+    assert_match "zlib", shell_output("#{bin}/spack list zlib")
+
+    # Set up configuration file and build paths
+    (testpath/"opt").mkpath
+    (testpath/"modules").mkpath
+    (testpath/"lmod").mkpath
+    (testpath/"stage").mkpath
+    (testpath/"cfg-store").mkpath
```

Comment:
The test is looking very nice. Maybe you can clean this one up a bit:
```suggestion
    %w[opt modules lmod stage cfg-store].each do |dir|
      (testpath/dir).mkpath
    end
```
I _think_ that works. Please test it locally first. I actually think there's an even simpler way to write this down, but my knowledge of Ruby stops here.

Reply:
Edited my suggestion. Think that one's better than a one-line `do` block.

Reply:
Yea, noticed some other code used `{}` instead of multiple lines. It seems to work.

Reply:
Looks like I missed a few directories that are created the first time spack is run -- fresh install and test immediately turned up a few cache directories that I added to the config file, and a junit-reports folder that I added a post install step to create (mkdir didn't seem to persist when used in the install step).